### DISCUSSION TITLE
New Cloud Resource Context for Issues and Performance

### DISCUSSION THREAD

**Discussion by smeubank** [getsentry/sentry#44406] [Launch Feedback]
### Description

We recently decided to add a new feature for our Python SDK, where we capture some basic information about where your service is running in AWS EC2 and GCP GCE. This should help developers know where and how their distributed services are actually hosted to speed up shipping fixes.

Possibly there are problems associated with a specific region? Or other specificities around your cloud hosting setup. You can't know it all we try to provide a little more context! :)

Docs [here](https://develop.sentry.dev/sdk/event-payloads/contexts/#cloud-resource-context)

![image](https://user-images.githubusercontent.com/47563310/218060645-2618d25f-d23a-4931-b07c-04ba81e5c9ee.png)

![image](https://user-images.githubusercontent.com/47563310/218060679-a2026723-6ce0-4751-8fd7-6305636d4616.png)

### Feedback
The original issue is [here](https://github.com/getsentry/sentry-python/issues/1820), and we will update this discussion once it is shipped

Would love to get your feedback, if there are other platforms/SDKs where you would like to see this feature, or other cloud hosting platforms.

**Comment by mwarkentin** [comment]
This is going to be huge for us as our infrastructure expands across zones and regions this year.. thanks so much!

**Comment by bruno-garcia** [comment]
I imagine this context is available to anything running on AWS, GCP, Azure. If this is proven valuable to customer, is the goal to add to guidelines to general SDK developers and add to Node, .NET, Java, etc?

**Reply by smeubank** [reply]
Hi @bruno-garcia , yep this could work for anything anywhere, we started small to fit the internal request but would like to see it scale out. If other platfroms have capacity to support or we see enough folks show interest.

@antonpirker added to [dev docs here ](https://docs.sentry.io/platforms/python/guides/rq/configuration/integrations/cloudresourcecontext/)

**Reply by antonpirker** [reply]
FYI right now it only works in AWS and GCP. 

This is the correct link to the spec of the context: https://develop.sentry.dev/sdk/event-payloads/contexts/#cloud-resource-context

As for implementation, all the cloud providers have internal API endpoints, where you can query information about the current context the code is running in: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/integrations/cloud_resource_context.py#L16-L25

I guess Azure has something similar and can be easily added if necessary.
